android-json-form-wizard/src/main/java/com/vijay/jsonwizard/interactors/JsonFormInteractor.java

```diff
@@ -180,15 +180,35 @@ public void fetchFields(List<View> viewsFromJson, String stepName, JsonFormFragm
         }
     }
 
-    private void fetchViews(List<View> viewsFromJson, String stepName, JsonFormFragment formFragment,
-                            String type, JSONObject jsonObject, CommonListener listener, Boolean popup) {
+    private void fetchViews(final List<View> viewsFromJson, final String stepName, final JsonFormFragment formFragment,
+                            final String type, final JSONObject jsonObject, final CommonListener listener, final Boolean popup) {
 
         try {
-            List<View> views = map
-                    .get(type)
-                    .getViewsFromJson(stepName, formFragment.getActivity(), formFragment, jsonObject, listener, popup);
-            if (views.size() > 0) {
-                viewsFromJson.addAll(views);
+            if (type.equals("tree") || type.equals("edit_text")) { ///some fields need to be loaded on the main thread TO-DO re-write
```

Comment:
Run the specific code that needs to be run in the UI thread in the factories and not here. This is a pattern that should be adopted